gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/GradeSummaryTablePanel.java

```diff
@@ -334,25 +334,33 @@ public boolean isVisible() {
 
 							gradeScore.add(sakaiRubricPreview);
 						} else {
-
 							gradeScore.add(
 									new Label("grade", FormatHelper.convertEmptyGradeToDash(FormatHelper.formatGradeForDisplay(rawGrade))));
 							gradeScore.add(new Label("outOf",
 									new StringResourceModel("label.studentsummary.outof", null, assignment.getPoints())));
 
 							final WebMarkupContainer sakaiRubricPreview = new WebMarkupContainer("sakai-rubric-student-button");
 							sakaiRubricPreview.add(AttributeModifier.append("display", "icon"));
-							sakaiRubricPreview.add(AttributeModifier.append("tool-id", RubricsConstants.RBCS_TOOL_GRADEBOOKNG));
-							sakaiRubricPreview.add(AttributeModifier.append("evaluated-item-id", assignment.getId() + "." + studentUuid));
 							sakaiRubricPreview.add(AttributeModifier.append("token", rubricsService.generateJsonWebToken(RubricsConstants.RBCS_TOOL_GRADEBOOKNG)));
 
 							if (!showingStudentView && (GradeSummaryTablePanel.this.getUserRole() == GbRole.INSTRUCTOR
 										|| GradeSummaryTablePanel.this.getUserRole() == GbRole.TA)) {
 								sakaiRubricPreview.add(AttributeModifier.append("instructor", true));
 							}
 
-							if (assignment.getId() != null) {
-								sakaiRubricPreview.add(AttributeModifier.append("entity-id", assignment.getId()));
+							if (assignment.isExternallyMaintained()) {
+								sakaiRubricPreview.add(AttributeModifier.append("tool-id", RubricsConstants.RBCS_TOOL_ASSIGNMENT));
+								String[] bits = assignment.getExternalId().split("/");
+								String assignmentId = bits[bits.length-1];
```

Comment:
no chance that this is null or doesn't have "/" ?

Reply:
Poss. I just left it like Bernardo did it. Just want the damn thing out of the door. I'll put a null check on it. It's an assignment tool reference.

Reply:
There's probably an assignment util for parsing the ref, actually. I'll check.